app/models/cohort.rb

```diff
@@ -35,11 +35,22 @@ def class_dates_until(last_date)
     (start_date..last_date).select { |date| !date.saturday? && !date.sunday? }
   end
 
+  def internships_sorted_by_interest(current_student)
+    self.internships.sort_by do |internship|
+      rating = current_student.find_rating(internship)
+      if rating
+        rating.interest
+      else
+        '0'
+      end
+    end
+  end
+
 private
 
   def import_code_reviews
     unless @importing_cohort_id.blank?
-      self.code_reviews = Cohort.find(@importing_cohort_id).deep_clone(include: { code_reviews: :objectives }).code_reviews
+      code_reviews = Cohort.find(@importing_cohort_id).deep_clone(include: { code_reviews: :objectives }).code_reviews
```

Comment:
So I added the `self` back on this line and `/spec/features/cohort_pages_spec.rb:40` suddenly passes. This is because `self.code_reviews` is really running `self.code_review=(...)`, which does ActiveRecord magic to associate all of those code reviews with the cohort (self). Whereas `code_reviews =` is just setting a variable called `code_reviews` equal to the cloned code reviews. (If you put a binding pry in there, you'll see that `code_reviews` will be an array with your cloned code reviews, but `code_reviews()` will be empty).

What a crazy bug. Another example of Ruby's syntactical sugar making things less expressive!

By the way, this is probably a good example of how we should try to only change code related to the feature we are trying to implement :)